### DISCUSSION TITLE
Using `prepare_cached`

### DISCUSSION THREAD

**Discussion by Zerowalker** [bikeshedder/deadpool#343] [Q&A] [answered]
I got a wrapper for the client and is wondering if using `prepared_cached` for all queries like this is fine:

```rust

#[derive(Clone)]
pub struct Db {
    pool: Pool,
}
impl Db {
    pub fn from(pool: Pool) -> Self {
        Db { pool }
    }
    async fn query_one(
        &self,
        statement: &str,
        params: &[&(dyn ToSql + Sync)],
    ) -> Result<Row, DbError> {
        let client = self.pool.get().await?;
        let stmt = client.prepare_cached(statement).await?;
        Ok(client.query_one(&stmt, params).await?)
    }
}
```

As there are limited hardcoded queries with just the parameters changing,
having them call `query_one`, would this generate/grab a prepared one,
thus making it potentially faster.
Or is this the wrong way to do this.

**Comment by bikeshedder** [comment, marked as answer]
`prepare_cached` should only be used for non-dynamic queries. So this is exactly what you're looking for. :+1: 

Sorry for the slow response. "Discussion" in GitHub doesn't show a unread counter and I miss new messages easily.

**Reply by Zerowalker** [reply]
No worries, was slow to respond as well.

Thanks, then I'm doing what I was hoping for:)!
